projects/npm-tools/packages/npm-scripts/src/utils/createTempFile.js

```diff
@@ -21,7 +21,11 @@ const BUILD_CONFIG = getMergedConfig('npmscripts', 'build');
  * An object with the `dispose` function (in case the caller wants to invoke it
  * before the end of the process) and the `filePath`.
  */
-function createTempFile(filename, content) {
+function createTempFile(filename, content, options) {
+	options = options || {
+		autoDelete: true,
+	};
```

Comment:
We don't have this documented in the guidelines anywhere yet, but a pattern we often use for dealing with options is [the one seen here](https://github.com/liferay/liferay-portal/blob/e4f606604ab9882875282c3167c4f56792697d1f/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/format_storage.es.js#L15-L43).

The problem with the `options = options || {...}` approach is twofold. One thing is that it would be better written using a default parameter:

```js
function createTempFile(filename, contents, options = {autoDelete: true}) {
    // ...
}
```

The other is that both the form in this PR and the form that uses a default parameter make it impossible to pass "partial" options. As in, just say I did `createTempFile(filename, content, {hello: 'friend'})`, `options.autoDelete` would end up being `undefined` (ie. falsey) even though the intent is clear that it should default to true.

The pattern that avoids this is:

```js
const DEFAULT_OPTIONS = {
    autoDelete: true,
};

function createTempFile(filename, content, options = {}) {
    const {autoDelete} = {...DEFAULT_OPTIONS, ...options};
}
```

Obviously not a deal-breaker for this PR, but I'm going to create an issue to remind us to document this pattern at some point.

Reply:
Here's the issue I made: https://github.com/liferay/liferay-frontend-projects/issues/409

I included there the "fancy" option that I forgot to mention of using destructuring with defaults:

```js
function createTempFile(filename, content, {autoDelete = true} = {}) {
    // ...
}
```

Reply:
True. I did it that way because it was the only option, so there was no possibility that we shadow others.

However, it is better to use the pattern you say always, even in this case, because otherwise we need to remember to change it if another option appears and, if we fail to do it, we cause a bug. :+1: